src/Hl7.Fhir.Support.Poco/Specification/Source/NpmPackageResolver.cs

```diff
@@ -0,0 +1,199 @@
+
+                var dependency = new PackageDependency(reference.Name ?? "", reference.Version);
+                if (reference.Found)
+                {
+                    var manifest = await scope.Project.ReadManifest();
+                    var fhirVersion = packageManifest.GetFhirVersion();
+                    if (fhirVersion is null)
+                    {
+                        throw new("Manifest doesn't contain a valid FHIR version");
+                    }
+                    manifest ??= ManifestFile.Create("temp", fhirVersion);
+                    if (manifest.Name == reference.Name)
+                    {
+                        throw new("Skipped updating package manifest because it would cause the package to reference itself.");
+                    }
+                    else
+                    {
+                        manifest.AddDependency(dependency);
+                        await scope.Project.WriteManifest(manifest);
+                        await scope.Restore();
+                    }
+                }
+            }
+
+        }
+
+        private Resource? toResource(string content)
+        {
+            var sourceNode = FhirJsonNode.Parse(content);
+            return TypedSerialization.ToPoco(sourceNode, _provider) as Resource;
+        }
+
+
+        ///<inheritdoc/>
+        public async Task<Resource?> ResolveByCanonicalUriAsync(string uri)
+        {
+            (var url, var version) = splitCanonical(uri);
+            var content = await _context.GetFileContentByCanonical(url, version, resolveBestCandidate: true);
+            return content is null ? null : toResource(content);
+        }
+
+
+        private static (string url, string version) splitCanonical(string canonical)
+        {
+            if (canonical.EndsWith("|"))
+                canonical = canonical.Substring(0, canonical.Length - 1);
+
+            var position = canonical.LastIndexOf('|');
+
+            return position == -1 ?
+                (canonical, "")
+                : (canonical.Substring(0, position), canonical.Substring(position + 1));
+        }
+
+
+        ///<inheritdoc/>
+        public async Task<Resource?> ResolveByUriAsync(string uri)
+        {
+            uri.SplitLeft('/').Deconstruct(out var resource, out var id);
+
+            if (resource == null || id is null)
+                return null;
+
+            var content = await _context.GetFileContentById(resource, id);
+            return content is null ? null : toResource(content);
+        }
+
+
+        private static PackageContext createContext(string folder, bool localCache = false)
+        {
+            string? cache_folder = localCache ? folder : null;
+            var cache = new DiskPackageCache(cache_folder);
+            var project = new FolderProject(folder);
+#pragma warning disable CS8625 // Cannot convert null literal to non-nullable reference type.
+            return new PackageContext(cache, project, null);
```

Comment:
So the third argument of the PackageContext constructor (`IPackageServer server`) should be made nullable

Reply:
Yes, in the package library. I'll fix that.